activation/activation.go

```diff
@@ -897,20 +911,23 @@ func (b *Builder) getPositioningAtx(
 	publish types.EpochID,
 	previous *types.ActivationTx,
 ) (types.ATXID, error) {
-	id, err := b.searchPositioningAtx(ctx, nodeID, publish)
+	ctxWithTimeout := ctx
+	var cancel context.CancelFunc
+
+	if b.poetCfg.PositioningATXSelectionTimeout > 0 {
+		ctxWithTimeout, cancel = context.WithTimeout(ctx, b.poetCfg.PositioningATXSelectionTimeout)
+		defer cancel()
+	}
```

Comment:
That unfortunately won't work like this.

Lets say the node finished PoST 10 minutes before the PoET round starts, then adding this timeout does nothing. The node will still try to pick a valid positioning ATX until right before the PoET round starts thereby possibly missing it.

```suggestion
	if b.poetCfg.PositioningATXSelectionTimeout > 0 {
		deadline, ok := ctx.Deadline()
		if ok {
			ctxWithTimeout, cancel = context.WithDeadline(ctx, deadline.Add(-b.poetCfg.PositioningATXSelectionTimeout))
			defer cancel()
		}
	}
```